src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java

```diff
@@ -1,526 +1,600 @@
+        }
+    }
+
+    @Test
+    public void testClosedConnection() throws SQLException {
+        try (Connection conn = getConnection()) {
+            conn.close();
+            assertTrue(!conn.isValid(0), TestResource.getResource("R_closedConnectionShouldBeInvalid"));
+
+            // getting shared timer on closed connection should fail
+            ((SQLServerConnection) conn).getSharedTimer();
+            fail(TestResource.getResource("R_noExceptionClosedConnection"));
+        } catch (SQLServerException e) {
+            assertEquals(e.getMessage(), TestResource.getResource("R_connectionIsClosed"),
+                    TestResource.getResource("R_wrongExceptionMessage"));
+        }
+        try (Connection conn = getConnection()) {
+            conn.close();
+            ((SQLServerConnection) conn).checkClosed();
+            fail(TestResource.getResource("R_noExceptionClosedConnection"));
+        } catch (SQLServerException e) {
+            assertEquals(e.getMessage(), TestResource.getResource("R_connectionIsClosed"),
+                    TestResource.getResource("R_wrongExceptionMessage"));
+        }
+    }
+
+    @Test
+    public void testSetTransactionIsolation() throws SQLException {
+        try (Connection conn = getConnection()) {
+            conn.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+            assertTrue(conn.getTransactionIsolation() == Connection.TRANSACTION_READ_COMMITTED,
+                    TestResource.getResource("R_expectedValue") + "Connection.TRANSACTION_READ_COMMITTED");
+            conn.setTransactionIsolation(Connection.TRANSACTION_READ_UNCOMMITTED);
+            assertTrue(conn.getTransactionIsolation() == Connection.TRANSACTION_READ_UNCOMMITTED,
+                    TestResource.getResource("R_expectedValue") + "Connection.TRANSACTION_READ_UNCOMMITTED");
+            conn.setTransactionIsolation(Connection.TRANSACTION_REPEATABLE_READ);
+            assertTrue(conn.getTransactionIsolation() == Connection.TRANSACTION_REPEATABLE_READ,
+                    TestResource.getResource("R_expectedValue") + "Connection.TRANSACTION_REPEATABLE_READ");
+            conn.setTransactionIsolation(Connection.TRANSACTION_SERIALIZABLE);
+            assertTrue(conn.getTransactionIsolation() == Connection.TRANSACTION_SERIALIZABLE,
+                    TestResource.getResource("R_expectedValue") + "Connection.TRANSACTION_SERIALIZABLE");
+            conn.setTransactionIsolation(SQLServerConnection.TRANSACTION_SNAPSHOT);
+            assertTrue(conn.getTransactionIsolation() == SQLServerConnection.TRANSACTION_SNAPSHOT,
+                    TestResource.getResource("R_expectedValue") + "SQLServerConnection.TRANSACTION_SNAPSHOT");
+        }
+    }
+
+    @Test
+    public void testNativeSQL() throws SQLException {
+        try (Connection conn = getConnection()) {
+            String nativeSql = conn.nativeSQL("SELECT @@version");
+            if (!(nativeSql instanceof String)) {

```

Comment:
Can you also verify the returned SQL value?

Reply:
added catching exception. This is just to execute any query to see if it succeeds it's typical in many tests.